### PR TITLE
use LOG_WARN for warnings instead of LOG_ERROR

### PR DESCRIPTION
**Pull request by VaibhaveS**
- Use LOG_WARN when processing warnings.

```
ives_interval=10&keepalives_count=60"
21:50:40 282611 NOTICE progress.c:110            Storing migration schema in JSON file "/tmp/pgcopydb/pgcopydb/schema.json"
21:50:40 282611 INFO   cli_clone_follow.c:547    STEP 2: dump the source database schema (pre/post data)
21:50:40 282611 INFO   pgcmd.c:469                /usr/bin/pg_dump -Fc --snapshot 00000007-00255D9A-1 --section pre-data --file /tmp/pgcopydb/pgcopydb/schema/pre.dump 'postgres://vaibhave%40testoid@testoid.postgres.database.azure.com/oid?keepalives=1&keepalives_idle=10&keepalives_interval=10&keepalives_count=60'
21:50:53 282611 WARN   string_utils.c:651        pg_dump: warning: WITH OIDS is not supported anymore (table "xyz")
21:50:54 282611 INFO   pgcmd.c:469                /usr/bin/pg_dump -Fc --snapshot 00000007-00255D9A-1 --section post-data --file /tmp/pgcopydb/pgcopydb/schema/post.dump 'postgres://vaibhave%40testoid@testoid.postgres.database.azure.com/oid?keepalives=1&keepalives_idle=10&keepalives_interval=10&keepalives_count=60'
```

### CODE REVIEW

File: src/bin/lib/subcommands.c/runprogram.h [hunk 1]
```diff
@@ -645,7 +645,7 @@ read_into_buf(Program *prog, int filedes, PQExpBuffer buffer, bool error)
 
 		if (prog->processBuffer)
 		{
-			(*prog->processBuffer)(temp_buffer, error);
+			(*prog->processBuffer)(temp_buffer, error, isWarning(temp_buffer));
```
Comment (dimitri) [inline src/bin/lib/subcommands.c/runprogram.h:648]: The lib/subcommands.c/ files should not be aware of anything that's not in the same directory. That's named a “modularity violation”. Please contain the implementation and call of the `isWarning` function within the src/bin/pgcopydb place. See other comments.

File: src/bin/pgcopydb/parsing_utils.c [hunk 2]
```diff
@@ -1080,3 +1080,12 @@ freeKeyVal(KeyVal *parameters)
...
 	parameters->count = 0;
 }
+
+/*
+ * returns true, if the string contains "warning"
+ */
+bool
+isWarning(const char* str)
+{
+	return strstr(str, "warning") != NULL;
```
Comment (dimitri) [inline src/bin/pgcopydb/parsing_utils.c:1090]: What if the buffer contains the user-provided string "warning", as in a column name or a table name for instance?

File: src/bin/pgcopydb/string_utils.c [hunk 3]
```diff
@@ -635,13 +635,13 @@ splitLines(char *buffer, char **linesArray, int size)
...
 void
-processBufferCallback(const char *buffer, bool error)
+processBufferCallback(const char *buffer, bool error, bool warning)
 {
 	char *outLines[BUFSIZE] = { 0 };
 	int lineCount = splitLines((char *) buffer, outLines, BUFSIZE);
 	int lineNumber = 0;
 
-	int logLevel = error ? LOG_ERROR : LOG_INFO;
+	int logLevel = warning ? LOG_WARN : (error ? LOG_ERROR : LOG_INFO);
```
Comment (dimitri) [inline src/bin/pgcopydb/string_utils.c:644]: I believe we should do buffer “analytics” here to decide if we're displaying at log level WARN or ERROR.

**Review by dimitri — CHANGES_REQUESTED** [review]
This PR has 2 main problems:
  - modularity violation (see comments)
  - parsing is too light to decide when we have a warning or an error (see comments)

**Comment by dimitri** [comment]
Also please avoid sharing pictures of text. That's the worst possible outcome. We want to be able to change the font size and colors of the text, copy/paste parts of it, and search/replace within the text. A screenshot makes all of that impossible, for zero benefits.

File: src/bin/pgcopydb/parsing_utils.c (hunk 2 above)
Comment (VaibhaveS) [reply src/bin/pgcopydb/parsing_utils.c:1090]: yeah, I had thought of it too. What do you think is the best way to filter warnings out?

How about this -

"**pg_dump: warning:**" and "**pg_restore: warning:**"

Comment (dimitri) [reply src/bin/pgcopydb/parsing_utils.c:1090]: Looks better already. I suppose it needs to be matched at the start of a line only?

File: src/bin/pgcopydb/parsing_utils.c [hunk 4]
```diff
@@ -1079,4 +1079,4 @@ freeKeyVal(KeyVal *parameters)
 	}
 
 	parameters->count = 0;
-}
+}
```
Comment (VaibhaveS) [inline src/bin/pgcopydb/parsing_utils.c:1082]: remove whitespace

**Review by dimitri — APPROVED** [review]

**Pull request merged**